docs/DOCUMENTATION.md

```diff
@@ -0,0 +1,872 @@
+
+
+## File structure
+
+In this section, one is able to find a description of the file structure and the contents of files
+
+#### scripts
+
+The scripts directory contains scripts which are used to for deployments. 
+
+The following are the different scripts which can be found in this directory
+
+1. accept-oracle-invitation.sh - This script can be used to accept an oracle invitation. This script takes in 3 parameters, the wallet name, the brand in and brand out. An example of a command to run this is ```./accept-oracle-invitation.sh $WALLET_NAME $BRAND_IN $BRAND_OUT```
+2. get-sdk-package-names.sh - This script was copied from the agoric-sdk repository and it is used to get the sdk package names.
+3. npm-audit-fix.sh - This script was copied from the agoric-sdk repository and it is used to fix issues with npm.
+4. provision-wallet.sh - This script can be used to provision a smart wallet. This script takes in one parameter, the wallet name. An example of a command to run this is ```./provision-wallet.sh $WALLET_NAME```
+
+#### src
+
+This directory includes all the source code. It is split into two other directories, <b>oracle</b> and <b>lib</b>.
+
+Furthermore, it contains the following two files which serve as an entry point to the middleware and monitoring script
+
+* <b>bin-middleware.js</b> - This serves as an entry point to the middleware by calling the middleware() function
+* <b>bin-monitor.js</b> - This serves as an entry point to the monitoring script by calling the getOraclesInvitations() and monitor() functions to first get the oracle invitation IDs and then starting the monitoring. 
+
+
+##### oracle
+
+This directory contains the following files
+
+1. <b>feeds.json</b> - This file contains the configuration for each feed in the oracle network. Each feed will have the following fields:
+  - <u>pollInterval</u> - The interval in seconds which needs to pass between each CL job
+  - <u>pushInterval</u> - The interval in seconds which needs to pass between each round creation on-chain
+  - <u>decimalPlaces</u> - The number of decimal places allowed for the price
+  - <u>priceDeviationPerc</u> - The price deviation percentage threshold on when to create a new round on-chain 
+The file should have the following structure
+```json
+{
+    "ATOM-USD" : {
+        "decimalPlaces": 6,
+        "pollInterval": 60,
+        "pushInterval": 600,
+        "priceDeviationPerc": 1 
+    },
+    "OSMO-USD" : {
+        "decimalPlaces": 6,
+        "pollInterval": 60,
+        "pushInterval": 600,
+        "priceDeviationPerc": 1 
+    }
+}
+```
+2. <b>helper.js</b> - This file contains helper functions used in middleware.js and monitor.js
+3. <b>middleware.js</b> - This file contains all the necessary code and functions for the middleware
+4. <b>monitor.js</b> - This file contains all the necessary code and functions for monitoring the oracle network
```

Comment:
Not really explaining much

Reply:
If code has comments, we can just export those and the functions used. Better than this imo

Reply:
It is there to give a brief explanation of what each file entails. Can be removed.